_posts/2023-01-16-Introducing-Identity.markdown

```diff
@@ -0,0 +1,44 @@
+# Introducing Identity for OpenSearch
+
+OpenSearch's existing access control features included in the Security plugin let administrators apply access control to indexes and cluster actions so that users have the right permissions to do their work and the cluster is protected from unwanted activity. However, the features for access control that are currently in place do have certain limitations that can make it hard to use them with other plugins.
+
+As the core OpenSearch project begins its shift away from a plugin model to a framework that utilizes extensions, those extensions, its legacy plugins, and the administrators who manage them will need mechanisms for controlling access that are more granular and able to cover a broader range of scenarios where effective access control is critical. We are creating/building out/developing a new suite of features that are designed to provide comprehensive access control to OpenSearch’s ecosystem, and we collectively call these new features Identity. 
+
+The main objectives for Identity include:
+
+- Mechanisms for OpenSearch, plugins, and extensions that can check permissions before attempting an action.
+- The ability to restrict plugins or extensions from performing actions unless they have been granted access.
+- Functionality that allows background tasks to run with the same access controls as interactive user requests.
+- The addition of new security boundaries inside OpenSearch that create conditions for a better defense-in-depth posture.
+
+Let’s look at these objectives one by one.
+
+### Enabling permissions checks 
+
+With any Using OpenSearch Dashboards your permissions you can start editing a Dashboard,and after hitting 'Save' seeing an error message "Dashboard ... was not saved. Error: Forbidden".  This scenario exists because Dashboards can not check if these commands are allowed or denied.
```

Comment:
This is a good example of the limitation of the current security model, but I think it needs to be reframed. 

Does it make sense to present an analogy here? In the current system there is no way to ask the question "Does this user have permission to perform this action?", instead what happens is that the user tries to perform and action and is blocked after initiating an action. In this case it leads to less-than-desirable UX because UI actions cannot be disabled when its known that a user does not have high enough privileges.

Reply:
Restructured, let me know what you think